spec/index/multi_index_spec.rb

```diff
@@ -35,6 +35,30 @@
           labels: [[0,0,1,1,2,2]])
       }.to raise_error
     end
+
+    context "create an MultiIndex with name" do
+      context "if no name is set" do
+        let(:mi) { Daru::MultiIndex.new(
+                    levels: [[:a,:b,:c], [:one, :two]],
+                    labels: [[0,0,1,1,2,2], [0,1,0,1,0,1]]) }
+        it { expect(mi.name).to be_nil }
+      end
+
+      context "correctly return the MultiIndex name" do
+        let(:mi) { Daru::MultiIndex.new(
+                  levels: [[:a,:b,:c], [:one, :two]],
+                  labels: [[0,0,1,1,2,2], [0,1,0,1,0,1]], name: ['n1', 'n2']) }
+        it { expect(mi.name).to eq ['n1', 'n2'] }
```

Comment:
When less number of `name` is set, then naming start from the last level to first and rest of the levels remain unnamed.

```
irb(main):004:0> mi
=> #<Daru::MultiIndex(6x2)>
  n1  n2
   a one
     two
   b one
     two
   c one
     two
irb(main):005:0> mi.name = ['k1']
=> ["k1"]
irb(main):006:0> mi
=> #<Daru::MultiIndex(6x2)>
      k1
   a one
     two
   b one
     two
   c one
     two
irb(main):007:0> mi.name = ['k1', 'k2']
=> ["k1", "k2"]
irb(main):008:0> mi
=> #<Daru::MultiIndex(6x2)>
  k1  k2
   a one
     two
   b one
     two
   c one
     two
```

ArgumentError: when more than number of `level` is passed.

```
irb(main):009:0> mi.name = ['k1', 'k2', 'k3']
=> ["k1", "k2", "k3"]
irb(main):010:0> mi
ArgumentError: negative argument
```

@zverok , is it okay?